src/app/shared/components/pagination/pagination.component.ts

```diff
@@ -0,0 +1,68 @@
+import { Component, EventEmitter, Input, OnChanges, Output } from '@angular/core';
+
+@Component({
+  selector: 'app-pagination',
+  templateUrl: './pagination.component.html',
+  styleUrls: ['./pagination.component.scss']
+})
+export class PaginationComponent implements OnChanges {
+  @Input() total: number = 0;
+  @Input() pageLimit: number = 5;
+
+  @Input() currentPage: number = 1;
+  @Output() currentPageChange = new EventEmitter<number>();
+
+  @Input() pageSize: number = 10;
+  @Output() pageSizeChange = new EventEmitter<number>();
+
+  @Output() paginationChange = new EventEmitter<PaginationChangeEvent>();
+
+  totalPages: number = 0;
+  pagesArr: number[] = [];
+
+  sliceStart: number = 0;
+  sliceEnd: number = this.pageLimit;
+  showingNumber: number = 0;
+
+  ngOnChanges(): void {
```

Comment:
You may use an if to compare what changed and then perform the needed actions as well.

Reply:
> You may use an if to compare what changed and then perform the needed actions as well.

Got it